modules/packages/LinearAlgebra.chpl

```diff
@@ -639,7 +639,39 @@ proc inner(A: [?Adom], B: [?Bdom]) {
   if Adom.size != Bdom.size then
     halt("Mismatched size in inner multiplication");
 
-  return + reduce(A[..]*B[..]);
+  var results: [0..#numLocales] etype = 0;
+  
+  coforall l in Locales do on l {
+    var localResult: etype = 0; 
+    const maxThreads = here.maxTaskPar;
```

Comment:
Question to reviewers: are we OK that this computation ignores --dataParTasksPerLocale and dataParIgnoreRunningTasks? They would be honored (hopefully) by the `+ reduce A*B` calculation.

Reply:
Although I think it is an edge case where you have an asynchronous tasks running in conjunction with this routine, it is definitely a good practice and consistent with the other similar places in the modules.

@wongalvis14 - See here for those variables: https://chapel-lang.org/docs/usingchapel/executing.html#controlling-degree-of-data-parallelism